mlir/lib/Dialect/SCF/Transforms/LoopSpecialization.cpp

```diff
@@ -191,6 +191,23 @@ static void rewriteAffineOpAfterPeeling(RewriterBase &rewriter, ForOp forOp,
   });
 }
 
+static void removeAffineOpInsideFirstIteration(RewriterBase &rewriter,
```

Comment:
I've removed this function in the revision because I feel it is not necessary. The `affine.min` can be kept as is in the peeled first iteration. And if it is useless, it normally can be canonicalized away as well as the forOp. The tests also validate this when `--canonicalize` is applied.